docs/protocol-basics.rst

```diff
@@ -153,10 +153,10 @@ block)
 
   * ``height`` is the height of the block it is in.
 
-3. Next, with mempool transactions in any order, append a similar
-string for those transactions, but where **height** is ``-1`` if the
-transaction has at least one unconfirmed input, and ``0`` if all
-inputs are confirmed.
+3. Next, with mempool transactions ordered by txid (with endianness
+same as displayed hex), append a similar string for those transactions,
+but where **height** is ``-1`` if the transaction has at least
+one unconfirmed input, and ``0`` if all inputs are confirmed.
```

Comment:
Hmm. I think it might be more natural to order them slightly differently.  Put all the 0's before all the -1's in the mempool case, at the end, and within the 0-group, order them by txid, and within the -1 group, order those by txid.  

This follows naturally form the "ordered by height" that precedes it, and just aesthetically seems nicer to me.

Reply:
The confirmed part of the history has increasing heights, so the mempool tx heights will not fit into that order (if they are at the end) in any case.
Also why would putting all the 0's before all the -1's be natural and not the other way around? I mean the confirmed part has increasing heights so why should the mempool part have decreasing? :P

Reply:
Your point is that it's an aesthetic arument, and it totally is.  It's arbitrary.  To me though the "confirmed parents" are "closer" in some abstract way to the actual confirmed tx's, as opposed to the unconfirmed ones that are "even less confirmed".  My argument is about how they are as a quality -- not their numeric values.  The tx's at height 3 are "more confirmed" than at height 10 than at height 600,000, and so on.  The tx's with no unconfirmed parents are less "unconfirmed" and more "confirmed" than the ones with unconfirmed parents.

To me it is the natural order based on their quality, not on the number itself.

Anyway touche -- I understand your point in that it's arbitrary.

But really think about my argument -- also it's what Fulcrum does now and I wouldn't have to change a bloody thing here!

Reply:
All right, I've changed the order to what you suggest.
I agree it's arbitrary, I don't really care what the order is, just that it is canonical.

Reply:
THANK YOU SO MUCH!!  

Electron Cash already expects this order as I described as does Fulcrum. I am very grateful for your openness here. Thanks!!